File: module_base/src/main/java/com/v/module_base/BaseActivity.java
package com.v.module_base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * activity基类
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getBaseLayoutId());
        initData();
        initPresenter();
        //隐藏标题栏
        initActionBar();
        //处理逻辑
        processLogical();
    }

    /**
     * 页面layoutId
     */
    public abstract int getBaseLayoutId();

    /**
     * 初始化数据
     */
    public abstract void initData();

    /**
     * 初始化Presenter
     */
    public abstract void initPresenter();

    /**
     * 初始化顶部ActionBar
     */
    public abstract void initActionBar();

    /**
     * 处理逻辑
     */
    public abstract void processLogical();
}

File: kotlin/src/main/java/com/v/kotlin/base/BaseActivity.kt
package com.v.kotlin.base

import android.os.Bundle
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity<P : IBasePresenter> : AppCompatActivity(){

    protected lateinit var mPresenter: P

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
        initData()

        mPresenter = createPresenter()

        //隐藏标题栏
        hideActionBar()

        //处理逻辑
        processLogical()
    }

    /**
     * 页面layoutId
     */
    abstract fun getLayoutId(): Int

    /**
     * 初始化数据
     */
    abstract fun initData()

    /**
     * 子类实现
     */
    abstract fun createPresenter(): P

    /**
     * 处理逻辑
     */
    abstract fun processLogical()

    /**
     * 隐藏actionbar
     */
    fun hideActionBar() {
        val actionBar: ActionBar? = supportActionBar
        actionBar?.hide()
    }

    /**
     * 显示actionbar
     */
    fun showActionBar() {
        supportActionBar?.hide()
    }


    override fun onDestroy() {
        super.onDestroy()
        mPresenter.unAttachView()
    }
}